MoltenVK/MoltenVK/Commands/MVKCmdTransfer.mm

```diff
@@ -1614,11 +1614,9 @@ static inline MTLSize mvkClampMTLSize(MTLSize size, MTLOrigin origin, MTLSize ma
     NSUInteger dstMTLBuffOffset = _dstBuffer->getMTLBufferOffset() + _dstOffset;
 
     // Copy data to the source MTLBuffer
-    MVKMTLBufferAllocation* srcMTLBufferAlloc = (MVKMTLBufferAllocation*)cmdEncoder->getCommandEncodingPool()->acquireMTLBufferAllocation(_dataSize);
+    MVKMTLBufferAllocation* srcMTLBufferAlloc = cmdEncoder->getCommandEncodingPool()->acquireMTLBufferAllocation(_dataSize);
     void* pBuffData = srcMTLBufferAlloc->getContents();
-    mlock(pBuffData, _dataSize);
     memcpy(pBuffData, _srcDataCache.data(), _dataSize);
-    munlock(pBuffData, _dataSize);
```

Comment:
First off, I assumed that would generally work, but after some more testing, it seems that just encoding the buffer in a command does not prevent Metal from discarding its contents prematurely*. I did initially consider simply adding a completion handler right there that delays this `munlock()` call, and that seems to work fine, but there's an edge case that I was worried about.
We might end up using two temporary buffers that come from a single `MTLBuffer` and share a page in it. If we did, `munlock`ing one of them would also unlock the other, or some part of it. If the commands were encoded in separate `MTLCommandBuffer`s that completed at different times, that would be bad, as the data in the other one could still be needed.
Since it looks like we have separate pools for each `VkCommandBuffer`, maybe there's no way this could happen. But I'm not sure - are there cases where we can get overlapping `MTLCommandBuffer` executions coming from a single `VkCommandBuffer`? As an aside, if we ever need to keep the contents of a temporary buffer across command buffer executions, it will complicate things even more.

Another reason that I preferred containing the memory locking in the allocator is that it is easy to forget about. I think it might be missing from `MVKInlineUniformBlockDescriptor`?

*Now I'm starting to think that maybe we should keep the buffer locked even if we don't initialize its contents and only use it while the `MTLCommandBuffer` is executing?